packages/patternfly-4/react-core/src/components/Nav/Nav.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import { NavVariants } from './NavVariants';
+
+const propTypes = {
+  /** Indicates the type of list. Only used for flat lists. */
+  variant: PropTypes.oneOf(Object.values(NavVariants)),
+  /** NavList elements */
+  children(props, propName, componentName) {
+    const prop = props[propName];
+
+    let error = null;
+    React.Children.forEach(prop, child => {
+      if (child.type.componentType !== 'NavList') {
+        error = new Error(`${componentName} children should be of type NavList.`);
+      }
+    });
+    return error;
+  },
+  /** Additional classes added to the Nav */
+  className: PropTypes.string,
+  /** Callback for updating when selection changes */
+  onSelect: PropTypes.func,
+  /** Callback for when a list is expanded or collapsed */
+  onToggle: PropTypes.func
+};
+
+const defaultProps = {
+  variant: 'default',
+  children: null,
+  className: '',
+  onSelect: () => undefined,
+  onToggle: () => undefined
+};
+
+const renderLists = (children, onSelect, onToggle) =>
+  React.Children.map(children, (childNode, index) =>
+    React.cloneElement(childNode, {
+      groupId: index,
+      onSelect,
+      onToggle
+    })
+  );
+
+class Nav extends React.Component {
+  // Callback from NavListItem
+  onSelect(groupId, itemId, event) {
+    this.props.onSelect({
+      itemId,
+      groupId,
+      event
+    });
+  }
+
+  // Callback from NavGroup
+  onToggle(groupId, isExpanded, event) {
+    this.props.onToggle({
+      groupId,
+      isExpanded,
+      event
+    });
+  }
+
+  render() {
+    const { variant, children, className, ...props } = this.props;
+
+    const getVariantStyle = type =>
+      ({
+        [NavVariants.default]: styles.navList,
+        [NavVariants.simple]: styles.navSimpleList,
+        [NavVariants.horizontal]: styles.navHorizontalList,
+        [NavVariants.tertiary]: styles.navTertiaryList
+      }[type]);
+
+    // variant style is only customizable if it's a single flat list
+    let variantStyle;
+    // Figure out if we have groups
+    let isGroup = false;
+    if (Array.isArray(children)) {
```

Comment:
Removing the variant style logic from the conditional. Groups do not have a variant style applied to them anyways so even if the user passes it in it won't have an effect. I still need to check if the user has passed in groups tho since the html structure is different for those.